pkg/fanal/handler/handler.go

```diff
@@ -75,5 +77,33 @@ func (m Manager) PostHandle(ctx context.Context, result *analyzer.AnalysisResult
 			return xerrors.Errorf("post handler error: %w", err)
 		}
 	}
+
+	// Overwrite package identifier info for system packages
+	if result != nil {
+		result.PackageInfos = overwritePackageIdentifiers(result.PackageInfos, result.OS)
+	}
+	if blob != nil {
+		blob.PackageInfos = overwritePackageIdentifiers(blob.PackageInfos, blob.OS)
+	}
+
 	return nil
 }
+
+// overwritePackageIdentifiers overwrite package identifiers on a given list of package info
+// This is useful when we want to overwrite package identifiers for OS packages
+// which original ones miss OS metadata info since they were generated in pkg (apk, rpm, etc.) analyzers
+func overwritePackageIdentifiers(pkgInfos []types.PackageInfo, os types.OS) []types.PackageInfo {
```

Comment:
I think it is better to use separate package.

Reply:
done at https://github.com/aquasecurity/trivy/pull/5439/commits/9a162b2d6031e362fad8a5b4ba493bbb65706984